pkg/destroy/openstack/openstack.go

```diff
@@ -125,6 +125,14 @@ func (o *ClusterUninstaller) Run() error {
 		o.Logger.Debugf("goroutine %v complete", res)
 	}
 
+	// we want to remove routers as the last thing as it requires detaching the
+	// FIPs and that will cause it impossible to track which FIPs are tied to
+	// LBs being deleted.
+	err = deleteRouterRunner(opts, o.Filter, o.Logger)
```

Comment:
Is there a risk the router is going to block the completion of any other resource deletion? If so you'll never reach this code. I think not but just want to make sure...

Reply:
Hmm, after looking at CI this is probably what's happening:
```
 INFO[2021-07-02T11:49:09Z] Deprovisioning cluster ...
{"component":"entrypoint","file":"prow/entrypoint/run.go:165","func":"k8s.io/test-infra/prow/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 45m0s timeout","severity":"error","time":"2021-07-02T11:39:06Z"} 
```

Reply:
If that's really what's happening you need to extract the deletion of the blocking resources to your second runner.

Reply:
Yep, I'll need to analyze that. :/

Reply:
Oh, seems like if it fails to remove an interface from the router it won't keep retrying so interface is left and blocks subnet removal. Easy fix but I need to check if it works.